src/mlpack/methods/ann/layer/bilinear_interpolation.hpp

```diff
@@ -0,0 +1,132 @@
+/**
+ * @file bilinear_interpolation.hpp
+ * @author Kris Singh and Shikhar Jaiswal
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license. You should have received a copy of the
+ * 3-clause BSD license along with mlpack. If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_METHODS_ANN_LAYER_BILINEAR_INTERPOLATION_HPP
+#define MLPACK_METHODS_ANN_LAYER_BILINEAR_INTERPOLATION_HPP
+
+#include <mlpack/prereqs.hpp>
+
+namespace mlpack {
+namespace ann /** Artificial Neural Network. */ {
+
+/**
+ * Definition and Implementation of the Bilinear Interpolation Layer.
+ *
+ * Bilinear Interpolation is an mathematical technique, primarily used for
+ * scaling purposes. It is an extension of linear interpolation, for
+ * interpolating functions of two variables on a rectangular grid. The key
+ * idea is to perform linear interpolation first in one direction (e.g., along
+ * x-axis), and then again in the other direction (i.e., y-axis), on four
+ * different known points in the grid. This way, we represent any arbitrary
+ * point, present within the grid, as a function of those four points.
+ *
+ * @tparam InputDataType Type of the input data (arma::colvec, arma::mat,
+ *         arma::sp_mat or arma::cube).
+ * @tparam OutputDataType Type of the output data (arma::colvec, arma::mat,
+ *         arma::sp_mat or arma::cube).
+ */
+template <
+    typename InputDataType = arma::mat,
+    typename OutputDataType = arma::mat
+>
+class BilinearInterpolation
+{
+ public:
+  //! Create the Bilinear Interpolation object.
+  BilinearInterpolation();
+
+  /**
+   * The constructor for the Bilinear Interpolation.
+   *
+   * @param inRowSize Number of input rows.
+   * @param inColSize Number of input columns.
+   * @param outRowSize Number of output rows.
+   * @param outColSize Number of output columns.
+   * @param depth Number of slices of input.
```

Comment:
Should we go with number of input slices? Also, what do you think should we remove the default depth = 1, people might miss to set the parameter.